cli/module_reload.go

```diff
@@ -165,34 +173,63 @@ func mutateModuleConfig(c *cli.Context, modules []ModuleMap, manifest moduleMani
 	if err != nil {
 		return nil, false, err
 	}
-	if foundMod == nil {
-		debugf(c.App.Writer, args.Debug, "module not found, inserting")
+
+	if foundMod != nil && getMapString(foundMod, "type") == string(rdkConfig.ModuleTypeRegistry) {
+		samePath, err := samePath(getMapString(foundMod, "reload_path"), absEntrypoint)
+		if err != nil {
+			return nil, dirty, err
+		}
+		reloadFlag := foundMod["reload_enabled"]
+		if samePath && reloadFlag == true {
+			debugf(c.App.Writer, args.Debug, "ReloadPath is up to date and ReloadEnabled, doing nothing")
+			return nil, dirty, err
+		}
 		dirty = true
-		newMod := ModuleMap(map[string]any{
-			"name":            localName,
-			"executable_path": absEntrypoint,
-			"type":            string(rdkConfig.ModuleTypeLocal),
-		})
-		modules = append(modules, newMod)
+		if samePath {
+			debugf(c.App.Writer, args.Debug, "ReloadPath is up to date, setting ReloadEnabled true")
+			foundMod["reload_enabled"] = true
+		} else {
+			debugf(c.App.Writer, args.Debug, "updating ReloadPath and ReloadEnabled")
+			foundMod["reload_path"] = absEntrypoint
+			foundMod["reload_enabled"] = true
+		}
 	} else {
-		if same, err := samePath(getMapString(foundMod, "executable_path"), absEntrypoint); err != nil {
-			debugf(c.App.Writer, args.Debug, "ExePath is right, doing nothing")
-			return nil, dirty, err
-		} else if !same {
-			dirty = true
-			debugf(c.App.Writer, args.Debug, "replacing entrypoint")
-			if getMapString(foundMod, "type") == string(rdkConfig.ModuleTypeRegistry) {
-				// warning: there's a chance of inserting a dupe name here in odd cases
-				warningf(c.App.Writer, "You're replacing a registry module. We're converting it to a local module. "+
-					"To revert this change, use your machine's history page on app.viam.com.")
-				foundMod["type"] = string(rdkConfig.ModuleTypeLocal)
-				foundMod["name"] = localName
-				delete(foundMod, "module_id")
-				// TODO(APP-5844): stop clearing this once backend no longer rejects; we will use it for revert
-				delete(foundMod, "version")
-			}
-			foundMod["executable_path"] = absEntrypoint
+		dirty = true
+		if foundMod == nil {
+			debugf(c.App.Writer, args.Debug, "module not found, inserting")
+		} else {
+			debugf(c.App.Writer, args.Debug, "found local module, inserting registry module")
 		}
+		newMod, err := createNewModuleMap(c, vc, localName, absEntrypoint)
+		if err != nil {
+			return nil, false, err
+		}
+		modules = append(modules, newMod)
 	}
 	return modules, dirty, nil
 }
+
+func createNewModuleMap(c *cli.Context, vc *viamClient, localName, entryPoint string) (ModuleMap, error) {
+	moduleID := moduleID{
+		prefix: vc.selectedOrg.Id,
+		name:   localName,
+	}
```

Comment:
Doing manual testing and I think this is the cause of the following error. The test case was running reload on a machine that doesn't have any modules in its config yet:

Error: Rpc error: code = InvalidArgument desc = queryID=47c20f51de0eaa7ee0424f0be7484a69: ModuleID ":michaellee1019_gpio-flicker_from_reload" is invalid. It must be "prefix:modulename" where prefix is either an orgid or a namespace.

I think lets source the moduleId from the manifest file. Reload command requires a meta.json anyways.